### ISSUE TITLE
[enhancement] Integrate USDA food databse in AAPS or nightscout

### ISSUE THREAD

**Issue by ragaimeena** [nightscout/AndroidAPS#2076]
The USDA has a free gigantic database for food including branded food etc..

[https://fdc.nal.usda.gov/index.html](url)
they provide an immediate api upon request or data can be downloaded in json format 
how can this be imlemented/integrated in the food option in either AAPS or nightscout?
sorry if this is a basic question but if there is an answer please guide me throuugh it
I appreciate it

**Comment by MilosKozak** [comment]
I don't think direct integration would help. UI for this would be complicated and during bolus (almost) nobody waste time

**Comment by Andries-Smit** [comment]
An alternative data source would be https://world.openfoodfacts.org/data

An alternative would be to integrate other food tracker apps trough APIs

Someone could make an tool to read food entries from tracker's API, when the users register the meal, the carbs could be read and then it can send the information to the Nightscout API. When the AAPS takes in the carbs from Nightscout , the SMB will handle the rest.

**Comment by ragaimeena** [comment]
> An alternative data source would be https://world.openfoodfacts.org/data
> 
> An alternative would be to integrate other food tracker apps trough APIs
> 
> Someone could make an tool to read food entries from tracker's API, when the users register the meal, the carbs could be read and then it can send the information to the Nightscout API. When the AAPS takes in the carbs from Nightscout , the SMB will handle the rest.

I agree. I think it will be a wonderful addition to this great app, which is already awsome. Having to go back and forth and look up the carbs etc.. in another app is inconvenient. I wish these databases can be imported into NS food editor which already syncs with AAPS, even better if it is all done via AAPS. I know you guys ared doing this great service for free, so it is much appreciated. Thank you

**Comment by duncanbarnes** [comment]
Perhaps another option would be for AndroidAPS to be able to receive intents if it can't already which would allow direct carbohydrate input websites supporting the WebShare API perhaps? The formats supported by WebShare are more limited, CSV seems a reasonable choice perhaps. That way a webapp implementing the dataset created by Open Food Facts could easily share into AAPS all the data for a meal etc, equally native Android apps could do the same thing.

Regrettably not an Android developer to assist with this, (although trying to find time to contribute some part finished code to cgm-remote-monitor), but for what its worth as another users opinion anyway it would be great for AAPS to be able to receive the data. That way the development burden is not on AAPS developers but it can integrate with other applications.

**Comment by ffsb42** [comment]
FYI, I had the opportunity to use IOS/Loop before switching to AAPS last week.... 

On IOS:

- I was using MyNetDiary app which has a very convenient way to enter food including carbs from a large curated database.
- All the data, including the carbs, get auomatically synced with Apple Health
- Loop read the same data in realtime from Apple Health achiving OP's request.

On Android: 
- the same application (MYNetDiary) exists but there is no such thing as apple health yet... apparently Google is releasing a beta application currently called "Health Connect by Android" which seems very similar. 
- it might make sense to look at this new "Health Conect" and integrate AAPS with it to read/share data in/out. 
- alternatively, it might be a good idea to extend the AAPS "automate" filter to grant the ability to trigger based on a notification string directly (if possible) this could provide an alternate way of caputuring such inputs from other apps... 
note that in both case, mynetdiary would have to a also integrate with health connect or with notifications for AAPS to be able to use it... but then again, maybe other apps will if they don't.

note that mynetdiary has a weird free/subscription/1time-pay model... I think the diabetic version si a 1time pay

**Comment by andyrozman** [comment]
"Health Conect" seems to be an option, but its not officially out yet... And like you said MYNetDiary would have to integrate into it first... or maybe they could broadcast sending and we could read this info in AAPS...
